File: LeetCode/Search-in-Rotated-Sorted-Array.cpp
1class Solution {
2public:
3    int search(vector<int>& nums, int target) {
4        int left = 0 ;
5        int right = nums.size()-1;
6
7        while (left <= right) {
8            int mid = (left + right)/2;
9
10            if (nums[mid ]== target ) {
11                return mid ;
12            }
13            if (nums[left ]<= nums[mid]) {
14                if (target >= nums[left] && target <= nums[mid]) {
15                    right = mid -1 ;
16                }
17                else {
18                    left = mid + 1;
19                }
20            }
21            else {
22                if (target >= nums[mid] && target <= nums[right]) {
23                    left = mid +1 ;
24                }
25                else {
26                  right = mid -1 ;
27                    }
28                }
29            }
30        
31        return -1;
32    }
33};
34

File: LeetCode/Search-in-Rotated-Sorted-Array.java
1class Solution {
2    public int search(int[] nums, int target) {
3        int n = nums.length;
4        int high = n-1;
5        int low = 0 ; 
6        
7        while (low <= high ){
8            int mid = (low + high)/2;
9            if (nums[mid]== target){
10               return  mid ;
11            }
12            if(nums[low]<= nums[mid]) {
13                if (nums[low]<= target && target <= nums[mid]){
14                    high = mid -1 ;
15
16                }
17                else low = mid +1 ;
18
19            }
20            else {
21                if (nums[mid]< target && target <= nums[high]){
22                    low = mid +1 ;
23                }
24                else high = mid -1 ;
25            }
26        }
27        return -1 ;
28    }
29}